pkg/agent/clientset.go

```diff
@@ -175,13 +171,13 @@ func (cs *ClientSet) syncOnce() error {
 	if err != nil {
 		return err
 	}
-	cs.serverCount = c.stream.serverCount
-	if err := cs.AddClient(c.stream.serverID, c); err != nil {
+	cs.serverCount = c.serverCount
```

Comment:
If cs.serverCount > 0 and cs.serverCount != c.serverCount, we should probably at least print a warning.

Reply:
Removed c.serverCount per your other comment :)